File: A2OJ/Ladder4_Codeforces_Div2A/007_ChatRoom/chatRoom.cpp
/*
 * Autor: Frederico Santos Gonçalves
 * Data: 19/01/2026
 *
 * Plataforma: A2OJ
 * Problema: Chat Room
 * Link: https://codeforces.com/problemset/problem/58/A
 * Dificuldade: 2
 */

#include <bits/stdc++.h>
#define SUCESSO (0)
using namespace std;
const string model = "hello";

int main() {
  string msg;
  getline(cin, msg);

  int posModel = 0;
  for(int i = 0; i < msg.length(); i++) {
    if(msg[i] == model[posModel]) posModel++;
    else continue;
  }

  if(posModel == model.length()) cout << "YES";
  else cout << "NO";

  return SUCESSO;
}

File: A2OJ/Ladder4_Codeforces_Div2A/007_ChatRoom/chatRoom.java
/*
 * Autor: Frederico Santos Gonçalves
 * Data: 21/01/2026
 *
 * Plataforma: A2OJ
 * Problema: Chat Room
 * Link: https://codeforces.com/problemset/problem/58/A
 * Dificuldade: 2
 */

import java.util.Scanner;

public class chatRoom {
  private static final Scanner scanner = new Scanner(System.in);
  private static final String model = "hello";

  public static void main(String[] args) {
    String msg = scanner.nextLine();

    int posModel = 0;
    for(int i = 0; i < msg.length(); i++) {
      if(msg.charAt(i) == model.charAt(posModel)) posModel++;
      if(posModel == model.length()) {
        System.out.println("YES");
        break;
      }
    }

    if(posModel < model.length()) System.out.println("NO");
  }
}
